src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -175,11 +238,82 @@ private SafeCurlHandle CreateRequestHandle(RequestCompletionSource state)
                 Interop.libcurl.curl_easy_setopt(requestHandle, CURLoption.CURLOPT_FOLLOWLOCATION, 1);
             }
 
-            // TODO: Handle headers, proxy and other options
+            SetProxyOptions(state, requestHandle);
+            // TODO: Handle headers and other options
 
             return requestHandle;
         }
 
+        private static void SetProxyOptions(RequestCompletionSource state, SafeCurlHandle requestHandle)
+        {
+            var requestUri = state.RequestMessage.RequestUri;
+            if (state.ProxyPolicy == CurlProxyUsePolicy.DoNotUseProxy)
+            {
+                Interop.libcurl.curl_easy_setopt(requestHandle, CURLoption.CURLOPT_PROXY, string.Empty);
+                return;
+            }
+
+            if ((state.ProxyPolicy == CurlProxyUsePolicy.UseDefaultProxy) || (state.Handler.Proxy == null))
+            {
+                return;
+            }
+
+            // It we reached here, Proxy != null AND ProxyPolixy = CustomProxy
+            if (state.Handler.Proxy.IsBypassed(requestUri))
+            {
+                Interop.libcurl.curl_easy_setopt(requestHandle, CURLoption.CURLOPT_PROXY, string.Empty);
+                return;
+            }
+
+            var proxyUri = state.Handler.Proxy.GetProxy(requestUri);
+            if (proxyUri == null)
+            {
+                return;
+            }
+
+            Interop.libcurl.curl_easy_setopt(requestHandle, CURLoption.CURLOPT_PROXYTYPE, CURLProxyType.CURLPROXY_HTTP);
+            Interop.libcurl.curl_easy_setopt(requestHandle, CURLoption.CURLOPT_PROXY, proxyUri.AbsoluteUri);
+            Interop.libcurl.curl_easy_setopt(requestHandle, CURLoption.CURLOPT_PROXYPORT, proxyUri.Port);
+            NetworkCredential credentials = GetCredentials(state.Handler.Proxy.Credentials, requestUri);
+            if (credentials != null)
+            {
+                if (string.IsNullOrEmpty(credentials.UserName))
+                {
+                    throw new ArgumentException(SR.net_http_argument_empty_string, "UserName");
+                }
+
+                var credentialText = string.Empty;
```

Comment:
```C#
string credentialText;
```
We're about to assign it to something, so there's no need to initialize it to empty.